worlds/ror2/Rules.py

```diff
@@ -7,27 +7,37 @@ class RiskOfRainLogic(LogicMixin):
     def _ror_has_items(self, player: int, amount: int) -> bool:
         count: int = self.item_count("Common Item", player) + self.item_count("Uncommon Item", player) + \
                      self.item_count("Legendary Item", player) + self.item_count("Boss Item", player) + \
-                     self.item_count("Lunar Item", player) + self.item_count("Equipment", player)
+                     self.item_count("Lunar Item", player) + self.item_count("Equipment", player) + \
+                     self.item_count("Dio's Best Friend", player)
         return count >= amount
 
 
 def set_rules(world: MultiWorld, player: int):
-    total_checks = world.total_locations[player]
     # divide by 5 since 5 levels (then commencement)
-    items_per_level = total_checks / 5
-    leftover = total_checks % 5
+    items_per_level = max(int(world.total_locations[player] / 5 / (world.item_pickup_step[player]+1)), 1)
 
-    set_rule(world.get_location("Level One", player),
-             lambda state: state._ror_has_items(player, items_per_level + leftover))
+    # lock item pickup access based on level completion
+    for i in range(1, items_per_level):
+        set_rule(world.get_location(f"Item Pickup {i}", player), lambda state: True)
```

Comment:
These all say `Item Pickup {x}` rather than `ItemPickup{x}` as is defined. Won't this cause a mismatch?

Reply:
I adjusted all cases of ItemPickup{x} to Item Pickup {x} so there's no mismatch. If you'd prefer it stay the old way this can be reverted though.

Reply:
As long as they match, it's fine. They _need_ to match.